infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactory.java

```diff
@@ -87,14 +91,20 @@ public BrokerEnvironmentFactory(
       String artifactsBrokerImage,
       String metadataBrokerImage,
       String pluginRegistryUrl,
+      String pluginRegistryInternalUrl,
       CertificateProvisioner certProvisioner) {
     this.cheWebsocketEndpoint = cheWebsocketEndpoint;
     this.brokerPullPolicy = brokerPullPolicy;
     this.authEnableEnvVarProvider = authEnableEnvVarProvider;
     this.machineTokenEnvVarProvider = machineTokenEnvVarProvider;
     this.artifactsBrokerImage = artifactsBrokerImage;
     this.metadataBrokerImage = metadataBrokerImage;
+    if (!isNullOrEmpty(pluginRegistryInternalUrl)) {
+      pluginRegistryUrl = pluginRegistryInternalUrl;
```

Comment:
Please don't modify input parameter here. This would be more straightforward imho.
```
this.pluginRegistryUrl = isNullOrEmpty(pluginRegistryInternalUrl) ? pluginRegistryUrl : pluginRegistryInternalUrl;
```